test/e2e/select.test.js

```diff
@@ -142,6 +141,49 @@ describe('vl-select', async () => {
     await assert.eventually.isTrue(select.hasValue('two'));
   });
 
+  it('als gebruiker kan ik een select dynamisch activeren met gegroepeerde opties', async () => {
+    const select = await vlSelectPage.getDynamicGroupingSelect();
+    await assert.eventually.isFalse(select.isDressed());
+    assert.equal((await select.groups()).length, 0);
+
+    await vlSelectPage.activeerDynamischeDataGrouping();
+
+    const selectMetData = await vlSelectPage.getDynamicGroupingSelect();
+    await selectMetData.open();
+    await assert.eventually.isTrue(selectMetData.isDressed());
+
+    const groups = await selectMetData.groups();
+    assert.equal(groups.length, 2);
+    await assert.eventually.equal(groups[0].getLabel(), 'Group 1');
+    await assert.equal(groups[0].options.length, 2);
+    await assert.eventually.equal(groups[0].options[0].getLabel(), 'Label one');
+    await assert.eventually.equal(groups[0].options[0].getValue(), 'one');
+
+    await assert.eventually.equal(groups[1].getLabel(), 'Group 2');
+    assert.equal(groups[1].options.length, 2);
+
+    await selectMetData.selectByValue('one');
+    await assert.eventually.equal(selectMetData.getSelectedValue(), 'one');
+  });
+
+  it('als gebruiker kan ik een de groepen van een undressed select opvragen', async () => {
+    const select = await vlSelectPage.getSelectUndressedGrouping();
+    await assert.eventually.isFalse(select.isDressed());
+
+    const groups = await select.groups();
+    assert.equal(groups.length, 2);
+    await assert.eventually.equal(groups[0].getLabel(), 'Group 1');
+    await assert.equal(groups[0].options.length, 2);
+    await assert.eventually.equal(groups[0].options[0].getLabel(), 'Label one');
+    await assert.eventually.equal(groups[0].options[0].getValue(), 'one');
+
+    await assert.eventually.equal(groups[1].getLabel(), 'Group 2');
+    assert.equal(groups[1].options.length, 2);
```

Comment:
Idem voorgaande opmerking, code kan best verwijderd of aangevuld worden.

Reply:
zie antwoord hierboven